src/WebJobs.Script.WebHost/StandbyManager.cs

```diff
@@ -44,16 +43,16 @@ public static bool IsWarmUpRequest(HttpRequest request)
                 request.Path.StartsWithSegments(new PathString($"/api/{WarmUpAlternateRoute}")));
         }
 
-        public static void Initialize(ScriptHostConfiguration config)
+        public static void Initialize(ScriptHostConfiguration config, ILoggerFactory standbyLoggerFactory)
         {
-            CreateStandbyFunctions(config.RootScriptPath, config.TraceWriter);
+            CreateStandbyFunctions(config.RootScriptPath, standbyLoggerFactory?.CreateLogger("Host.Standby"));
         }
 
-        private static void CreateStandbyFunctions(string scriptPath, TraceWriter traceWriter)
+        private static void CreateStandbyFunctions(string scriptPath, ILogger logger)
         {
             lock (_syncLock)
             {
-                traceWriter.Info($"Creating StandbyMode placeholder function directory ({scriptPath})");
+                logger?.LogInformation($"Creating StandbyMode placeholder function directory ({scriptPath})");
```

Comment:
I know that it is an existing problem that everywhere we're using loggers we seem to be guarding against null. However we need to be able to assume we have the ability to log at some point and that we can rely on these being non-null. Otherwise we're not logging.

In this case we had a non-null logger before. Why can it be null now?

Reply:
I can remove all the null-checks. Now that `ILogger` is the only way to log, you're right, we should make sure that it's there.

Reply:
I removed all these null-conditional operators. It actually found one bug where we were passing in a null LoggerFactory due to some ordering issues. I've filed an issue as it will require some more refactoring and may just need to be done when we implement DI: https://github.com/Azure/azure-webjobs-sdk-script/issues/2223